gemma-core/src/main/java/ubic/gemma/core/apps/AffyProbeCollapseCli.java

```diff
@@ -31,59 +32,70 @@
  * Purely a testing tool, to turn Affy individual probes (by probeset) into collapsed sequences. This is what happens
  * internally when we add sequences to a platform, but this makes it possible to see (and check) the sequences first
  * before committing to them.
- * 
+ * <p>
  * Should probably be in GemmaAnalysis but that is badly broken.
- * 
+ * <p>
  * You just run this like
- * 
+ * <p>
  * $GEMMACMD affyCollapse [filename]
- * 
+ * <p>
  * It doesn't handle the regular argument setup, wasn't worth the trouble. Generates FASTA format but easy to change.
- * 
+ *
  * @author paul
  */
 public class AffyProbeCollapseCli extends ArrayDesignSequenceManipulatingCli {
-    public static void main( String[] args ) {
+
+    private String affyProbeFileName;
+
+    public static int main( String[] args ) {
         AffyProbeCollapseCli d = new AffyProbeCollapseCli();
-        executeCommand( d, args );
+        return d.executeCommand( args );
     }
 
     /*
      * (non-Javadoc)
-     * 
+     *
      * @see ubic.gemma.core.util.AbstractCLI#getCommandName()
      */
     @Override
     public String getCommandName() {
         return "affyCollapse";
     }
 
+    @Override
+    protected void buildOptions() {
+        super.buildOptions();
+        addOption( Option.builder( "affyProbeFile" )
+                .hasArg()
+                .desc( "Affymetrix probe file to use as input" )
+                .required().build() );
+    }
+
+    @Override
+    protected void processOptions() {
+        super.processOptions();
+        affyProbeFileName = this.getOptionValue( "affyProbeFile" );
+    }
+
     /*
      * (non-Javadoc)
-     * 
+     *
      * @see ubic.gemma.core.util.AbstractCLI#doWork(java.lang.String[])
      */
     @Override
-    protected Exception doWork( String[] args ) {
+    protected void doWork() throws IOException {
 
         // parse
         AffyProbeReader apr = new AffyProbeReader();
-        try {
-            apr.parse( args[0] );
```

Comment:
This change will require some adjustment in the documentation as we now have a proper argument for the affy probe file.

I think that this was intended to be a positional argument, @ppavlidis is this supported with addOption?

Reply:
boy I can't remember why that is written that way, another relic. Making it a proper option is good but let's shorten the option name to "probeFile" rather than affyProbeFile, and yes, the docs on the wiki will need to be updated. But we also probably never use that CLI any more so not a big deal